x-pack/test/functional_endpoint/services/endpoint_policy.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+import {
+  CreateAgentConfigResponse,
+  CreateDatasourceResponse,
+} from '../../../plugins/ingest_manager/common';
+import { Immutable } from '../../../plugins/endpoint/common/types';
+import { factory as policyConfigFactory } from '../../../plugins/endpoint/common/models/policy_config';
+
+const INGEST_API_ROOT = '/api/ingest_manager';
+const INGEST_API_AGENT_CONFIGS = `${INGEST_API_ROOT}/agent_configs`;
+const INGEST_API_AGENT_CONFIGS_DELETE = `${INGEST_API_AGENT_CONFIGS}/delete`;
+const INGEST_API_DATASOURCES = `${INGEST_API_ROOT}/datasources`;
+const INGEST_API_DATASOURCES_DELETE = `${INGEST_API_DATASOURCES}/delete`;
+
+/**
+ * Holds information about the test resources created to support an Endpoint Policy
+ */
+export interface PolicyTestResourceInfo {
+  /** The Ingest agent configuration created */
+  agentConfig: Immutable<CreateAgentConfigResponse['item']>;
+  /** The Ingest datasource created and added to agent configuration.
+   * This is where Endpoint Policy is stored.
+   */
+  datasource: Immutable<CreateDatasourceResponse['item']>;
+  /** will clean up (delete) the objects created (agent config + datasource) */
+  cleanup: () => Promise<void>;
+}
+
+export function EndpointPolicyTestResourcesProvider({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  return {
+    /**
+     * Creates an Ingest Agent Configuration and adds to it the Endpoint Datasource that
+     * stores the Policy configuration data
+     */
+    async createPolicy(): Promise<PolicyTestResourceInfo> {
+      // FIXME: Refactor after https://github.com/elastic/kibana/issues/64822 is fixed. `isInitialized` setup below should be deleted
+      // Due to an issue in Ingest API, we first create the Fleet user. This avoids the Agent Config api throwing a 500
+      const isFleetSetupResponse = await supertest
+        .get('/api/ingest_manager/fleet/setup')
+        .set('kbn-xsrf', 'xxx')
+        .expect(200);
+      if (!isFleetSetupResponse.body.isInitialized) {
+        await supertest
+          .post('/api/ingest_manager/fleet/setup')
```

Comment:
should there be another const here?

Reply:
@paul-tavares - I think a service is good for now. I'm not sure what the tradeoffs are between doing it this way and using something like es_archiver. I'll research and let us know.

Reply:
@kevinlog I did not create one because I hope to remove this soon - so its all centralized to this block of code.

Reply:
@ai - I just realized (even after speaking with you on the phone) that I should really add a better description to this PR that expand a little on why this -vs- esArchiver is being used. I will do that next.